src/Microsoft.SqlTools.SqlCore/Scripting/AsyncScriptAsScriptingOperation.cs

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core;
+using Microsoft.SqlTools.SqlCore.Scripting.Contracts;
+
+namespace Microsoft.SqlTools.SqlCore.Scripting
+{
+    public class AsyncScriptAsScriptingOperation
+    {
+        public static async Task<string> GetScriptAsScript(ScriptingParams parameters, AccessToken? accessToken)
+        {
+            var scriptAsOperation = new ScriptAsScriptingOperation(parameters, accessToken?.Token);
+            TaskCompletionSource<string> scriptAsTask = new TaskCompletionSource<string>();
+
+            scriptAsOperation.CompleteNotification += (sender, args) =>
+            {
+                if (args.HasError)
+                {
+                    scriptAsTask.SetException(new Exception(args.ErrorMessage));
+                }
+                scriptAsTask.SetResult(scriptAsOperation.ScriptText);
+            };
+
+            scriptAsOperation.ProgressNotification += (sender, args) =>
+            {
+                scriptAsTask.SetException(new Exception(args.ErrorMessage));
```

Comment:
Did you copy this from somewhere? Not sure why a "progress notification" is always being handled as an error.